agenta-cli/agenta/client/Readme.md

```diff
@@ -1,3 +1,123 @@
+Currently the models are manually copied from the backend code. This needs to change.
+
+# Generate Backend
+
+To generate the client code using Fern, follow the steps below. 
+
+1. Next ensure you have installed Fern by executing the command;
+    ```bash
+        $ npm install -g fern-api
+    ```
+2. Execute this command to initialize Fern to import and use the OpenAPI spec;
+
+> To use an OpenAPI spec, you can pass in the filepath or URL.
+> We'll be using a url to the openapi.json for Aggenta at https://cloud.agenta.ai.
+
+    ```bash
+        fern init --openapi https://cloud.agenta.ai/api/openapi.json
+    ```
+
+1. Add Security Definition to the openapi.json
+   
+    At this stage, you should have a folder named "fern" with the following directory;
+
+        ```bash
+            fern/
+                ├─ fern.config.json
+                ├─ generators.yml
+                openapi/
+                    ├─ openapi.yml
+        ```
+
+    - Go to the openapi.json.
+    - Go to components and add the following new schema;
+        ```bash
+            securitySchemes:
+                APIKeyHeader:
+                type: apiKey
+                in: header
+                name: Authorization
+        ```
+    - At the last empty line in the bottom of the file, add this;
+        ```bash
+            security:
+            - APIKeyHeader: []
+        ```
+    
+
+4. Run `fern check` to validate the OpenAPI spec and resolve any errors, if there are any. ( You can ignore the conflicting endpoints warnings. )
+   
+5. Add the Fern Python SDK;
+   ```bash
+        fern add fern-python-sdk
+    ```
+
+6. Go to the generators.yml, which would look like this;
+
+    ```bash
+        default-group: local
+        groups:
+        local:
+            generators:
+            - name: fernapi/fern-typescript-node-sdk
+                version: 0.7.2
+                output:
+                  location: local-file-system
+                  path: ../generated/typescript
+            - name: fernapi/fern-python-sdk
+                version: 0.6.0
+    ```
+
+    Replace the following;
+
+    ```bash
+        - name: fernapi/fern-typescript-node-sdk
+          version: 0.7.2
+    ```
+
+    with this;
+
+    ```bash
+       - name: fernapi/fern-python-sdk
+         version: 0.6.0
+    ```
+
+    Also, change the path and path name to reflect the path name of python sdj appropriately and change the path to where you want the generate code to be. 
+    For example change it from this `path: ../generated/typescript` to this path: `../backend`
+
+    Now your generators.yml should look like this;
+    ```bash
+        default-group: local
+        groups:
+        local:
+            generators:
+            - name: fernapi/fern-python-sdk
+                version: 0.6.0
+                output:
+                  location: local-file-system
+                  path: ../backend
+    ```
+
+7. Generate the client code
+   
+    ```bash
+        fern generate
```

Comment:
When running the `fern generate` command, the backend code is not generated. Instead, it only displays the following in the terminal:

![image](https://github.com/Agenta-AI/agenta/assets/55067204/a102c6b3-ddd7-4e32-8969-a36e12e16b74)
![image](https://github.com/Agenta-AI/agenta/assets/55067204/e2b25bc1-b7f7-4320-a282-2bd3b2412a1c)

I have followed all the steps in the readme. Do you think there is something I might be doing wrong?

Reply:
Check Slack for a message I'll send you.